File: RedBoxMan_JavaFX/src/redboxman_javafx/RedBoxManRenderer.java
package redboxman_javafx;

import java.io.InputStream;
import java.util.ArrayList;
import javafx.application.Application;
import javafx.geometry.Point2D;
import javafx.scene.Group;
import javafx.scene.Scene;
import javafx.scene.canvas.Canvas;
import javafx.scene.canvas.GraphicsContext;
import javafx.scene.image.Image;
import javafx.scene.paint.Paint;
import javafx.stage.Stage;

/**
 *
 * @author McKillaGorilla
 */
public class RedBoxManRenderer extends Application {
    Canvas canvas;
    GraphicsContext gc;
    ArrayList<Point2D> imagesRedBoxManLocations;
    ArrayList<Point2D> shapesRedBoxManLocations;
    ArrayList<Point2D> shapesRedRoundManLocations;
    Image redBoxManImage;
    
    @Override
    public void start(Stage primaryStage) {
	// INIT THE DATA MANAGERS
	imagesRedBoxManLocations = new ArrayList<>();
	shapesRedBoxManLocations = new ArrayList<>();
	shapesRedRoundManLocations = new ArrayList<>();
	
	// LOAD THE RED BOX MAN IMAGE
        InputStream str = getClass().getResourceAsStream("/RedBoxMan.png");
	redBoxManImage = new Image(str);
	
	// MAKE THE CANVAS
	canvas = new Canvas();
	canvas.setStyle("-fx-background-color: cyan");
	gc = canvas.getGraphicsContext2D();

	// PUT THE CANVAS IN A CONTAINER
	Group root = new Group();
	root.getChildren().add(canvas);
	
	canvas.setOnMouseClicked(e->{
	    if (e.isShiftDown()) {
		shapesRedBoxManLocations.add(new Point2D(e.getX(), e.getY()));
		render();
	    }
	    else if (e.isControlDown()) {
		imagesRedBoxManLocations.add(new Point2D(e.getX(), e.getY()));
		render();
	    }else if(e.isAltDown()){
		shapesRedRoundManLocations.add(new Point2D(e.getX(), e.getY()));
		render();
	    }
	    else {
		clear();
	    }
	});
	
	// PUT THE CONTAINER IN A SCENE
	Scene scene = new Scene(root, 800, 600);
	canvas.setWidth(scene.getWidth());
	canvas.setHeight(scene.getHeight());

	// AND START UP THE WINDOW
	primaryStage.setTitle("Red Box Man Renderer");
	primaryStage.setScene(scene);
	primaryStage.show();
    }
    
    public void clearCanvas() {
	gc.clearRect(0, 0, canvas.getWidth(), canvas.getHeight());
    }
    
    public void clear() {
	shapesRedBoxManLocations.clear();
	imagesRedBoxManLocations.clear();
	render();
    }
    
    public void render() {
	clearCanvas();
	for (int i = 0; i < shapesRedBoxManLocations.size(); i++) {
	    renderShapeRedBoxMan(shapesRedBoxManLocations.get(i));
	}
	for (int j = 0; j < imagesRedBoxManLocations.size(); j++) {
	    renderImageRedBoxMan(imagesRedBoxManLocations.get(j));
	}
	for (int k = 0; k < shapesRedRoundManLocations.size(); k++){
		renderShapeRedRoundMan(shapesRedRoundManLocations.get(k));
	}
    }
    
    public void renderShapeRedBoxMan(Point2D location) {
	String headColor = "#DD0000";
	String outlineColor = "#000000";
	int headW = 115;
	int headH = 88;
    
	// DRAW HIS RED HEAD
        gc.setFill(Paint.valueOf(headColor));
	gc.fillRect(location.getX(), location.getY(), headW, headH);
        gc.beginPath();
	gc.setStroke(Paint.valueOf(outlineColor));
	gc.setLineWidth(1);
	gc.rect(location.getX(), location.getY(), headW, headH);
	gc.stroke();
	
	// AND THEN DRAW THE REST OF HIM
    }

    public void renderShapeRedRoundMan(Point2D location){
	    //Draw Body
	   double bodyX = location.getX() + (25);
	   double bodyY = location.getY() + (70);
	   int bodyW = 60;
	   int bodyH = 30;
	   gc.setFill(Paint.valueOf("#000000"));
	   gc.fillRect(bodyX, bodyY, bodyW, bodyH);
	   gc.fillRect(bodyX + 7.5, bodyY + 20, 45, 20);
	   gc.fillRect(bodyX - 2, bodyY + 35, 10, 10);
	   gc.fillRect(bodyX + bodyW - 8, bodyY +35, 10, 10);



	    String headColor = "#40E0D0";
	    String outlineColor = "#000000";
	    int headW = 115;
	    int headH = 88;

	    //Draw his red head
	    gc.setFill(Paint.valueOf(headColor));
	    gc.fillOval(location.getX(), location.getY(), headW, headH);
	    gc.beginPath();
	    gc.setStroke(Paint.valueOf(outlineColor));
	    gc.setLineWidth(1);
	    gc.strokeOval(location.getX(), location.getY(), headW, headH);
	    gc.stroke();

	    //Draw his eyes
	    String eyeColor = "#FFFF00";
	    int eyeW = 30;
	    int eyeH = 20;

	    double leftX = location.getX() + (15);
	    double leftY = location.getY() + (25);
	    double rightX = location.getX() + (65);
	    double rightY= leftY;

	    gc.setFill(Paint.valueOf(eyeColor));
	    gc.fillOval(leftX, leftY, eyeW, eyeH);
	    gc.fillOval(rightX, rightY, eyeW, eyeH);
	    gc.beginPath();
	    gc.strokeOval(leftX, leftY, eyeW, eyeH);
	    gc.strokeOval(rightX, rightY, eyeW, eyeH);
	    gc.stroke();

	    //Draw pupils
	    String black = "#000000";
	    int pupilW = 10;
	    int pupilH = 5;
	    double pupilLeftX = leftX + 9;
	    double pupilLeftY = leftY + 8;
	    double pupilRightX = rightX + 9;
	    double pupilRightY = pupilLeftY;

	    gc.setFill(Paint.valueOf(black));
	    gc.fillOval(pupilLeftX, pupilLeftY, pupilW, pupilH);
	    gc.fillOval(pupilRightX, pupilRightY, pupilW, pupilH);
	    gc.beginPath();


	   //Draw mouth
	   double mouthX = location.getX() + 20;
	   double mouthY = location.getY() + 60;
	   int mouthW = 70;
	   int mouthH = 7;
	   gc.fillRect(mouthX, mouthY, mouthW, mouthH);
	   gc.beginPath();

	   

    }
    
    public void renderImageRedBoxMan(Point2D location) {
	gc.drawImage(redBoxManImage, location.getX(), location.getY());
    }

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
	launch(args);
    }
    
}

File: RedBoxMan_HTML5/public_html/js/RedBoxManRenderer.js
/* 
 * This JavaScript file provides methods for clearing the
 * canvas and for rendering Red Box Man.
 */

var canvas;
var gc;
var canvasWidth;
var canvasHeight;
var imageLoaded;
var mousePositionRendering;
var redBoxManImage;
var mouseX;
var mouseY;
var imagesRedBoxManLocations;
var shapesRedBoxManLocations;

function Location(initX, initY) {
    this.x = initX;
    this.y = initY;
}

function init() {
    // GET THE CANVAS SO WE CAN USE IT WHEN WE LIKE
    canvas = document.getElementById("red_box_man_canvas");
    gc = canvas.getContext("2d");
        
    // MAKE SURE THE CANVAS DIMENSIONS ARE 1:1 FOR RENDERING
    canvas.width = canvas.offsetWidth;
    canvas.height = canvas.offsetHeight;
    canvasWidth = canvas.width;
    canvasHeight = canvas.height;
    
        // FOR RENDERING TEXT
    gc.font="32pt Arial";

    // LOAD THE RED BOX MAN IMAGE SO WE CAN RENDER
    // IT TO THE CANVAS WHENEVER WE LIKE
    redBoxManImage = new Image();
    redBoxManImage.onload = function() {
	imageLoaded = true;
    }
    redBoxManImage.src = "./images/RedBoxMan.png";
    
    // BY DEFAULT WE'LL START WITH MOUSE POSITION RENDERING ON
    mousePositionRendering = true;
    
    // HERE'S WHERE WE'LL PUT OUR RENDERING COORDINATES
    imagesRedBoxManLocations = new Array();
    shapesRedBoxManLocations = new Array();
}

function processMouseClick(event) {
    updateMousePosition(event);
    var location = new Location(mouseX, mouseY);
    if (event.shiftKey) {
	shapesRedBoxManLocations.push(location);
	render();
    }
    else if (event.ctrlKey) {
	if (imageLoaded) {
	    imagesRedBoxManLocations.push(location);
	    render();
	}
    }
    else {
	clear();
    }
}

function clearCanvas() {
    gc.clearRect(0, 0, canvasWidth, canvasHeight);
}

function clear() {
    shapesRedBoxManLocations = [];
    imagesRedBoxManLocations = [];
    clearCanvas();
}

function updateMousePosition(event) {
    var rect = canvas.getBoundingClientRect();
    mouseX = event.clientX - rect.left;
    mouseY = event.clientY - rect.top;
    render();
}

function renderShapesRedRoundMan(location){
 //Draw Body
	   var bodyX = location.x + (25);
	   var bodyY = location.y + (70);
	   var bodyW = 60;
	   var bodyH = 30;
	   gc.fillStyle = "#000000";
	   gc.fillRect(bodyX, bodyY, bodyW, bodyH);
	   gc.fillRect(bodyX + 7.5, bodyY + 20, 45, 20);
	   gc.fillRect(bodyX - 2, bodyY + 35, 10, 10);
	   gc.fillRect(bodyX + bodyW - 8, bodyY +35, 10, 10);
	   gc.beginPath();

	    var headColor = "#40E0D0";
	    var outlineColor = "#000000";
	    var headW = 115;
	    var headH = 88;

	    //Draw his red head
	    gc.fillStyle = headColor;
	    gc.fillRect(location.x, location.y, headW, headH);
	    gc.beginPath();
	    gc.strokeStyle = outlineColor;
	    gc.lineWidth = 1;
	    gc.rect(location.x, location.y, headW, headH);
	    gc.stroke();

	    //Draw his eyes
	    var eyeColor = "#FFFF00";
	    var eyeW = 30;
	    var eyeH = 20;

	    var leftX = location.x + (15);
	    var leftY = location.y + (25);
	    var rightX = location.x + (65);
	    var rightY= leftY;

	    gc.fillStyle = eyeColor;
	    gc.fillRect(leftX, leftY, eyeW, eyeH);
	    gc.fillRect(rightX, rightY, eyeW, eyeH);
	    gc.beginPath();
	    gc.rect(leftX, leftY, eyeW, eyeH);
	    gc.rect(rightX, rightY, eyeW, eyeH);
	    gc.stroke();

	    //Draw pupils
	    var black = "#000000";
	    var pupilW = 10;
	    var pupilH = 5;
	    var pupilLeftX = leftX + 9;
	    var pupilLeftY = leftY + 8;
	    var pupilRightX = rightX + 9;
	    var pupilRightY = pupilLeftY;

	    gc.fillStyle = black;
	    gc.fillRect(pupilLeftX, pupilLeftY, pupilW, pupilH);
	    gc.fillRect(pupilRightX, pupilRightY, pupilW, pupilH);
	    gc.beginPath();


	   //Draw mouth
	   var mouthX = location.x + 20;
	   var mouthY = location.y + 60;
	   var mouthW = 70;
	   var mouthH = 7;
	   gc.fillRect(mouthX, mouthY, mouthW, mouthH);
	   gc.beginPath();
}

function renderShapesRedBoxMan(location) {
    var headColor = "#DD0000";
    var outlineColor = "#000000";
    var headW = 115;
    var headH = 88;
    
    // DRAW HIS RED HEAD
    gc.fillStyle = headColor;
    gc.fillRect(location.x, location.y, headW, headH);
    gc.beginPath();
    gc.strokeStyle = outlineColor;
    gc.lineWidth = 1;
    gc.rect(location.x, location.y, headW, headH);
    gc.stroke();
    
    // AND THEN DRAW THE REST OF HIM
}

function renderImageRedBoxMan(location) {
    gc.drawImage(redBoxManImage, location.x, location.y);
}

function renderMousePositionInCanvas(event) {
    if (mousePositionRendering) {
	gc.strokeText("(" + mouseX + "," + mouseY + ")", 10, 50);    
    }
}

function toggleMousePositionRendering() {
    mousePositionRendering = !mousePositionRendering;
}

function render() {
    clearCanvas();
    for (var i = 0; i < shapesRedBoxManLocations.length; i++) {
	var location = shapesRedBoxManLocations[i];
	renderShapesRedRoundMan(location);
	//renderShapesRedBoxMan(location);
    }
    for (var j = 0; j < imagesRedBoxManLocations.length; j++) {
	var location = imagesRedBoxManLocations[j];
	renderImageRedBoxMan(location);
    }
    renderMousePositionInCanvas();
}